src/ucp/core/ucp_context.h

```diff
@@ -115,6 +115,9 @@ typedef struct ucp_context {
     ucp_tl_md_t                   *tl_mds;    /* Memory domain resources */
     ucp_rsc_index_t               num_mds;    /* Number of memory domains */
 
+    ucp_tl_md_t                   **mem_type_tl_mds; /* mem type MD resources */
```

Comment:
would add more info here - that these are the MDs which detect non-host memory
BTW, maybe just use static allocation? e.g instead of malloc, just declare:
`ucp_tl_md_t                   *mem_type_tl_mds[UCT_MD_MEM_TYPE_LAST]`?
or:
`ucp_rsc_index_t        mem_type_tl_mds[UCT_MD_MEM_TYPE_LAST]`
- to store md_index instead of pointer, as done in other places in UCP which want to refer to a particular MD

Reply:
fixed.  used  ucp_rsc_index_t array